### PR TITLE
she -> [they] in Dining Philosophers

### PR DESCRIPTION
**Pull request by steveklabnik**
Also, when checking for common gendered words elsewhere, I found one 'he', moved to 'they' as well.

https://github.com/rust-lang/rust/pull/25640#issuecomment-104304643

### CODE REVIEW

File: src/doc/trpl/dining-philosophers.md [hunk 1]
```diff
@@ -176,7 +178,7 @@ that `new()` function, it would look like this:
 #     name: String,
 # }
 fn main() {
-    let p1 = Philosopher { name: "Baruch Spinoza".to_string() };
+    let p1 = Philosopher { name: "Judith Butler".to_string() };
     let p2 = Philosopher { name: "Gilles Deleuze".to_string() };
     let p3 = Philosopher { name: "Karl Marx".to_string() };
     let p4 = Philosopher { name: "Friedrich Nietzche".to_string() };
```
Comment (vberger) [inline src/doc/trpl/dining-philosophers.md]: Shouldn't this one be changed to "Emma Goldman" as the others ?

Comment (steveklabnik) [inline src/doc/trpl/dining-philosophers.md]: oh my, i did a search-replace, the mis-spelling missed it, thanks

File: src/doc/trpl/dining-philosophers.md [hunk 2]
```diff
@@ -2,26 +2,28 @@
...
-1965, but we’ll use the version from [this paper][paper] by Tony Hoare in 1985.
+1965, but we’ll use a lightly adapted version from [this paper][paper] by Tony
+Hoare in 1985.
 
 [paper]: http://www.usingcsp.com/cspbook.pdf
 
 > In ancient times, a wealthy philanthropist endowed a College to accommodate
-> five eminent philosophers. Each philosopher had a room in which she could
+> five eminent philosophers. Each philosopher had a room in which they could
 > engage in her professional activity of thinking; there was also a common
```
Comment (killercup) [inline src/doc/trpl/dining-philosophers.md]: s/in her profe/in their profe

File: src/doc/trpl/dining-philosophers.md [hunk 3]
```diff
@@ -2,26 +2,28 @@
...
-> in her own chair, picked up her own fork on her left, and plunged it into the
-> spaghetti. But such is the tangled nature of spaghetti that a second fork is
-> required to carry it to the mouth. The philosopher therefore had also to pick
-> up the fork on her right. When she was finished she would put down both her
-> forks, get up from her chair, and continue thinking. Of course, a fork can be
-> used by only one philosopher at a time. If the other philosopher wants it, she
-> just has to wait until the fork is available again.
+> was constantly replenished. A philosopher was expected to spend most of
+> their time thinking; but when they felt hungry, they went to the dining
+> room, sat down in thier own chair, picked up their own fork on her left,
```
Comment (killercup) [inline src/doc/trpl/dining-philosophers.md]: s/thier/their

File: src/doc/trpl/dining-philosophers.md [hunk 4]
```diff
@@ -2,26 +2,28 @@
...
+> and plunged it into the spaghetti. But such is the tangled nature of
+> spaghetti that a second fork is required to carry it to the mouth. The
+> philosopher therefore had also to pick up the fork on their right. When
+> they was finished they would put down both their forks, get up from her
```
Comment (killercup) [inline src/doc/trpl/dining-philosophers.md]: s/they was/they were

Comment (killercup) [inline src/doc/trpl/dining-philosophers.md]: s/from her/from their

Comment (steveklabnik) [inline src/doc/trpl/dining-philosophers.md]: thanks :)

**Comment by steveklabnik** [comment]
proofreading: it is hard :sweat:

**Comment by andrew-d** [comment]
I've stayed out of this discussion (and the other threads), but a major :+1: for the care and sensitivity that you and everyone else has put into this :cake:

File: src/liballoc/rc.rs [hunk 5]
```diff
@@ -144,7 +144,7 @@
 //!     // At the end of the method, gadget_owner, gadget1 and gadget2 get
 //!     // destroyed. There are now no strong (`Rc<T>`) references to the gadgets.
 //!     // Once they get destroyed, the Gadgets get destroyed. This zeroes the
-//!     // reference count on Gadget Man, so he gets destroyed as well.
+//!     // reference count on Gadget Man, they get destroyed as well.
```
Comment (crazymykl) [inline src/liballoc/rc.rs]: I think the 'so' makes the causal relation clearer.

**Pull request merged**